File: python/asmt3/asmt3_starter/basictester.py
from wordoccurrences import WordOccurrences

topdir = ".\\test_dir\\Music\\Beatles"

wo = WordOccurrences( topdir )
'''
print("distinct words: " + str(wo.distinctWords()))
print("total occurrences: " + str(wo.totalOccurrences()))
print("total occurrences of love: " + str(wo.totalOccurrences("love")))
print("total occurrences of ob-la-di: " + str(wo.totalOccurrences("ob-la-di")))
print("total occurrences of detest: " + str(wo.totalOccurrences("detest")))

print("total occurrences of that's in Money.txt: "
                + str(wo.totalOccurrences("that's", ".\\test_dir\\Music\\Beatles\\Money.txt")))

print("total occurrences of ob-la-di in Money.txt: "
                + str(wo.totalOccurrences("ob-la-di", ".\\test_dir\\Music\\Beatles\\Money.txt")))

print("total occurrences of the in Funny.txt: "
                + str(wo.totalOccurrences("the", ".\\test_dir\\Music\\Beatles\\Funny.txt")))

print("-------------")
'''
# Uncomment below to see how well the __str__ function works

print( wo ) 

File: java/asmt2/src/BasicTester.java
import word_occurrences.WordOccurrences;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.io.PrintWriter;

public class BasicTester {

    public static void main(String[] args) throws IOException {
        test1();
        test2();
    }

    public static void test1() throws IOException {

        System.out.println("Running test1");

        // You may have to change the file path to get it to work on your computer
        WordOccurrences occ = new WordOccurrences(new File("tests//wisdom.txt"));
        System.out.println(occ);
        System.out.println("distinct words: " + occ.distinctWords());
        System.out.println("total occurrences: " + occ.totalOccurrences());
        System.out.println();
    }

    public static void test2() throws IOException {

        System.out.println("------------------");
        System.out.println("Running test2");

        // You may have to change the file path to get it to work on your computer
        WordOccurrences occ = new WordOccurrences(new File("tests"));

        System.out.println("distinct words: " + occ.distinctWords());
        System.out.println("total occurrences: " + occ.totalOccurrences());
        System.out.println("total occurrences of love: " + occ.totalOccurrencesOfWord("love"));
        System.out.println("total occurrences of ob-la-di: " + occ.totalOccurrencesOfWord("ob-la-di"));
        System.out.println("total occurrences of detest: " + occ.totalOccurrencesOfWord("detest"));

        System.out.println("total occurrences of that's in Money.txt: "
                + occ.totalOccurrencesOfWordInFile("that's", "test_dir\\Music\\Beatles\\Money.txt"));

        System.out.println("total occurrences of ob-la-di in Money.txt: "
                + occ.totalOccurrencesOfWordInFile("ob-la-di", "test_dir\\Music\\Beatles\\Money.txt"));

        System.out.println("total occurrences of the in Funny.txt: "
                + occ.totalOccurrencesOfWordInFile("the", "test_dir\\Music\\Beatles\\Funny.txt"));

        System.out.print(occ);
        File outFile = new File("output.txt");
        PrintWriter writer = new PrintWriter(new FileOutputStream(outFile));
        writer.write(occ.toString());
    }

}
